src/main/java/ee/tuleva/onboarding/comparisons/fundvalue/retrieval/GlobalStockIndexRetriever.java

```diff
@@ -0,0 +1,187 @@
+        } finally {
+            try {
+                ftpClient.close();
+            } catch(IOException ignored) {}
+        }
+        return extractValuesFromRecords(records, startDate, endDate);
+    }
+
+    private List<FundValue> extractValuesFromRecords(List<DailyRecord> records, LocalDate startDate, LocalDate endDate) {
+        ArrayList<FundValue> fundValues = new ArrayList<>();
+
+        for(LocalDate date = startDate; date.isBefore(endDate) || date.isEqual(endDate); date = date.plusDays(1)) {
+            String monthID = date.format(DateTimeFormatter.ofPattern("yyyyMM"));
+            int dayOfMonth = date.getDayOfMonth();
+
+            DailyRecord record = records.stream()
+                .filter(r -> r.secID.equals(secID) && r.monthID.equals(monthID))
+                .findAny()
+                .orElse(null);
+
+            if(record == null)
+                continue;
+
+            String value = record.values.get(dayOfMonth - 1);
+            if(!value.isEmpty()) {
+                fundValues.add(new FundValue(GlobalStockIndexRetriever.KEY, date, new BigDecimal(value)));
+            }
+        }
+        return fundValues;
+    }
+
+    private void pushDailyRecord(List<DailyRecord> records, DailyRecord record) {
+        int recordIdx = records.indexOf(record);
+        if(recordIdx != -1) {
+            records.get(recordIdx).update(record);
+        } else {
+            records.add(record);
+        }
+    }
+
+    private Optional<DailyRecord> findInZip(InputStream stream, String secId) throws IOException {
+        try(ZipInputStream zipStream = new ZipInputStream(stream)) {
+            // Just get one file and we are done.
+            ZipEntry entry = zipStream.getNextEntry();
+
+            if(entry != null) {
+                return findInCSV(zipStream, secId);
+            } else {
+                return Optional.empty();
+            }
+        }
+    }
+
+    private Optional<DailyRecord> findInCSV(InputStream stream, String secId) throws IOException {
+        try(BufferedReader reader = new BufferedReader(new InputStreamReader(stream, UTF_8))) {
+            Stream<String> lines = reader.lines();
+            return lines.map(this::parseLine)
+                .filter((DailyRecord record) -> record.secID.equals(secId))
+                .findFirst();
+        }
+    }
+
+    private DailyRecord parseLine(String line) {
+        String[] parts = line.split(",", -1);
+        int partLength = parts.length;
+        return new DailyRecord(parts[0], parts[1], Arrays.asList(parts).subList(2, partLength));
+    }
+
+    private static class DailyRecord {
+        @Getter private String secID;
+        @Getter private String monthID;
+        @Getter private ArrayList<String> values;
+
+        DailyRecord(String _secID, String _monthID, Collection<String> _values) {
+            secID = _secID;
+            monthID = _monthID;
+            values = new ArrayList<>(_values);
+        }
```

Comment:
Use the `this` keyword instead of underscores to avoid name collisions.
```
    class DailyRecord {
        @Getter private String securityId;
        @Getter private String monthId;
        @Getter private List<String> values;

        DailyRecord(String securityId, String monthId, List<String> values) {
            this.securityId = securityId;
            this.monthId = monthId;
            this.values = values;
        }

        ...
```